### DISCUSSION TITLE
cannot change the state of a quasar button by toggling its disabled prop with a boolean value

### DISCUSSION THREAD

**Discussion by AhmedHamdy-B** [quasarframework/quasar#16359] [General - Components / Directives / etc]
I am trying to toggle the disable prop of a quasar input by clicking an edit button, this button basically changes the value of the "clientDetailsEditToggle" ref to be the opposite of its boolean value, if true sets to false.
![image](https://github.com/quasarframework/quasar/assets/56585994/ed1da221-97b8-4964-aa9e-cc372c169b8b)

but when I try to do something like that it shows this error
![image](https://github.com/quasarframework/quasar/assets/56585994/464a01bb-7da0-4edf-8497-4d8b5dc2626b)

is there anything I can do that or any workaround?
note: I am using vee-validate and I want to bind this value to multiple inputs so when the user toggles it he can edit the multiple inputs and switch its state

**Comment by metalsadman** [comment]
clientDetailsEditToggle might not be a reactive, you are not showing it tho. it should be a reactive property resulting to a boolean.

**Reply by AhmedHamdy-B** [reply]
it is reactive actually it is a vue3 ref() with default value of false

**Reply by metalsadman** [reply]
please provide a reproduction

**Reply by rstoenescu** [reply]
You're essentially supplying `clientDetailsEditToggle` to the `disable` prop, but that is the Vue reference to that respective component.

```
<q-input
  ref="clientDetailsEditToggle"
  :disable="clientDetailsEditToggle"
```

First problem is that you're not supplying a Boolean value. Second, you might force Vue to end up in an endless loop by playing with the Vue ref this way.

**Reply by AhmedHamdy-B** [reply]
I removed the line  ref="clientDetailsEditToggle" which and tried to change the value of the  clientDetailsEditToggle with a function which is the clientDetailsEditToggleFn 
    const clientDetailsEditToggleFn = () => {
      clientDetailsEditToggle.value = !clientDetailsEditToggle.value;
    };
    but it doesnt change the disable behavior to true and false, it just sets it to the default value that was set which is false.
    
    
![image](https://github.com/quasarframework/quasar/assets/56585994/7a7d1b48-a99b-4222-954e-065ec5e14b62)

**Comment by metalsadman** [comment]
@AhmedHamdy-B there's definitely something wrong somewhere in your code, but like I said above, make a reproduction, image of your code doesn't help much in this case. see https://codepen.io/metalsadman/pen/qBLobrL
